# Additional wrangling + finding ad-hoc answers

In [2]:
# Import libraries
import pandas as pd
import numpy as np
import os

In [3]:
# Import data set
df = pd.read_csv(r'/Users/OldBobJulia/Desktop/CF/Course/6. Advanced Analytics and Dashboard Design/Berlin Airbnb Analysis/02 Data/Prepared data/listing_derivedcolumns.csv')

In [4]:
df.head()

Unnamed: 0       neighbourhood_group  latitude  longitude        room_type  \
0           0                     Mitte  52.54433   13.39761     Private room   
1           1                     Mitte  52.53305   13.40394  Entire home/apt   
2           2                    Pankow  52.53471   13.41810  Entire home/apt   
3           3    Tempelhof - Schöneberg  52.49884   13.34940     Private room   
4           4  Friedrichshain-Kreuzberg  52.51163   13.45289  Entire home/apt   

   price  minimum_nights  number_of_reviews  reviews_per_month  \
0   28.0              60                 18               0.21   
1   74.0              90                141               2.20   
2   90.0              62                147               1.14   
3   29.0               7                 27               0.28   
4   79.0               7                135               1.02   

   calculated_host_listings_count  availability_365 Price category  \
0                               3               101      Low price   
1                               6               357      Low price   
2                               1               254   Middle price   
3                               1               285      Low price   
4                               1                 0      Low price   

  Rental availability        Host type  
0         Middle term  Commercial host  
1           Long term  Commercial host  
2           Long term     Private host  
3           Long term     Private host  
4          Short term     Private host

In [5]:
# Drop Unnamed: 0 
df = df.drop(columns = ['Unnamed: 0'])

In [6]:
df.shape

(19095, 13)

In [7]:
# Adjust neighbourhood names
df['neighbourhood_group'] = df['neighbourhood_group'].replace({'Charlottenburg-Wilm.': 'Charlottenburg-Wilmersdorf', 'Tempelhof - Schöneberg': 'Tempelhof-Schöneberg', 'Treptow - Köpenick': 'Treptow-Köpenick', 'Steglitz - Zehlendorf': 'Steglitz-Zehlendorf', 'Marzahn - Hellersdorf': 'Marzahn-Hellersdorf'})

In [8]:
df.head()

neighbourhood_group  latitude  longitude        room_type  price  \
0                     Mitte  52.54433   13.39761     Private room   28.0   
1                     Mitte  52.53305   13.40394  Entire home/apt   74.0   
2                    Pankow  52.53471   13.41810  Entire home/apt   90.0   
3      Tempelhof-Schöneberg  52.49884   13.34940     Private room   29.0   
4  Friedrichshain-Kreuzberg  52.51163   13.45289  Entire home/apt   79.0   

   minimum_nights  number_of_reviews  reviews_per_month  \
0              60                 18               0.21   
1              90                141               2.20   
2              62                147               1.14   
3               7                 27               0.28   
4               7                135               1.02   

   calculated_host_listings_count  availability_365 Price category  \
0                               3               101      Low price   
1                               6               357      Low price   
2                               1               254   Middle price   
3                               1               285      Low price   
4                               1                 0      Low price   

  Rental availability        Host type  
0         Middle term  Commercial host  
1           Long term  Commercial host  
2           Long term     Private host  
3           Long term     Private host  
4          Short term     Private host

In [8]:
# Create path
path = r'/Users/OldBobJulia/Desktop/CF/Course/6. Advanced Analytics and Dashboard Design/Berlin Airbnb Analysis'

In [9]:
# Export
df.to_csv(os.path.join(path, '02 Data','Prepared Data', 'listing_derived_wrangled.csv'))

In [9]:
df['neighbourhood_group'].value_counts(dropna=False)

Friedrichshain-Kreuzberg      4197
Mitte                         4173
Pankow                        2981
Neukölln                      2608
Charlottenburg-Wilmersdorf    1567
Tempelhof-Schöneberg          1371
Treptow-Köpenick               620
Lichtenberg                    612
Steglitz-Zehlendorf            415
Reinickendorf                  271
Marzahn-Hellersdorf            142
Spandau                        138
Name: neighbourhood_group, dtype: int64

In [11]:
# Compare private vs commercial hosts 
df['Host type'].value_counts(dropna=False)

Private host       15893
Commercial host     3202
Name: Host type, dtype: int64

In [15]:
# Check income mean
df.groupby('Host type').agg({'price': ['mean', 'min', 'max', 'sum']})

price                           
                       mean  min     max           sum
Host type                                             
Commercial host  102.075578  9.0  2575.0  3.268460e+05
Private host      66.042416  8.0  4000.0  1.049612e+06

In [14]:
# See availability grouped by host type
df.groupby('Host type').agg({'availability_365': ['mean', 'min', 'max']})

availability_365         
                            mean min  max
Host type                                
Commercial host       215.420050   0  365
Private host           66.259234   0  365

In [19]:
# Compare room types 
df['room_type'].value_counts(dropna=False)

Entire home/apt    10636
Private room        8042
Shared room          249
Hotel room           168
Name: room_type, dtype: int64

In [20]:
df.groupby('room_type').agg({'availability_365': ['mean', 'min', 'max', 'sum']})

availability_365                  
                            mean min  max      sum
room_type                                         
Entire home/apt       102.090071   0  365  1085830
Hotel room            260.303571   0  365    43731
Private room           71.667496   0  365   576350
Shared room           148.281124   0  365    36922

In [23]:
# Import additional libraries
import seaborn as sns
import matplotlib

In [28]:
# How many entire apartments are available more than 90 days? Create subset. 
long_term = df[df['availability_365']>=90]

In [29]:
long_term.head()

neighbourhood_group  latitude  longitude        room_type  price  \
0                 Mitte  52.54433   13.39761     Private room   28.0   
1                 Mitte  52.53305   13.40394  Entire home/apt   74.0   
2                Pankow  52.53471   13.41810  Entire home/apt   90.0   
3  Tempelhof-Schöneberg  52.49884   13.34940     Private room   29.0   
7                Pankow  52.54813   13.40366  Entire home/apt   70.0   

   minimum_nights  number_of_reviews  reviews_per_month  \
0              60                 18               0.21   
1              90                141               2.20   
2              62                147               1.14   
3               7                 27               0.28   
7              90                 26               0.26   

   calculated_host_listings_count  availability_365 Price category  \
0                               3               101      Low price   
1                               6               357      Low price   
2                               1               254   Middle price   
3                               1               285      Low price   
7                               3               238      Low price   

  Rental availability        Host type  
0         Middle term  Commercial host  
1           Long term  Commercial host  
2           Long term     Private host  
3           Long term     Private host  
7           Long term  Commercial host

In [30]:
long_term.shape

(6329, 13)

In [31]:
long_term['room_type'].value_counts(dropna=False)

Entire home/apt    4026
Private room       2023
Hotel room          144
Shared room         136
Name: room_type, dtype: int64

**- We have 15893 private hosts and 3202 commercial hosts**

**- The mean incomes are 66 (private) vs 102 EUR (commercial) per night**

**- The mean availability of rooms throughout the year are 66 (private) vs 215 (commercial)**